### PR TITLE
Add PDF rendering via PDFObject

### PR DESCRIPTION
**Pull request by silverwind**
Use [PDFObject](https://pdfobject.com/) to embed PDFs, replacing our outdated PDF.js copy we vendor (the last non-webpack vendoring).

[Commit 1](https://github.com/go-gitea/gitea/pull/24086/commits/673e0263da64b72565ff59b990ab1b8e87271872) is the PDFObject integration
[Commit 2](https://github.com/go-gitea/gitea/pull/24086/commits/9336f5769d54445bba0e16776164f6a2fe2c32ac) is the removal of PDF.js

Screenshot 2023-05-27 at 09 57 52
Screenshot 2023-05-27 at 10 12 50

Fallback for unsupporting browsers (most mobile ones, except Firefox Mobile):

Screenshot 2023-05-27 at 09 43 34

### CODE REVIEW

**Comment by silverwind** [comment]
Text selection is working but there are a few other bugs remaining like dynamic scale factor calculation etc. The pdf.js API is very low-level and almost undocumented, so it takes some effort.

**Comment by silverwind** [comment]
Thought I do wonder whether this TextLayer business is warranted and maintainable. Parts of this API depends on CSS that Mozilla keeps changing, so I don't think they see it as part of the API contract. Maybe it's best to leave the PDF integration to the simple Canvas renderer which API ought to be very stable and this is what GitLab and GitHub are currently doing.

**Comment by wxiaoguang** [comment]
Is it possible to drop the PDF viewer package, just let browser render? IIRC you also proposed similar idea before.

**Comment by silverwind** [comment]
What we could do thought, is use PDFObject and fall back to the pdf.js canvas rendering (no selection etc).

**Comment by silverwind** [comment]
Actually I might just end up going with PDFObject and a fallback button like "Click here to view the PDF document". pdf.js turns out to be a real pain to have as a dependency because of its `canvas` dependency.

**Comment by silverwind** [comment]
Updated to use PDFObject and updated OP message with screenshots, so it's ready for review again.

**Comment by silverwind** [comment]
Pushed the PDF.js removal as well in a separate commit, so review only commit 1.

**Review by delvh — APPROVED** [review]

File: templates/repo/settings/lfs_file.tmpl [hunk 1]
```diff
@@ -19,7 +19,7 @@
 					{{else if .IsPlainText}}
 						<pre>{{if .FileContent}}{{.FileContent | Safe}}{{end}}</pre>
 					{{else if not .IsTextFile}}
-						<div class="view-raw ui center">
+						<div class="view-raw ui center{{if .IsPDFFile}} gt-p-0{{end}}">
```
Comment (delvh) [inline templates/repo/settings/lfs_file.tmpl]: I haven't quite understood this change here and in the next file.
Why only when it is a PDF file?

Comment (silverwind) [reply templates/repo/settings/lfs_file.tmpl]: Could maybe done for other types as well, I need to check.

Comment (silverwind) [reply templates/repo/settings/lfs_file.tmpl]: Removed the margin from CSS and did a few more cleanups to view raw in https://github.com/go-gitea/gitea/pull/24086/commits/f9e8b38712bff75deae29f41ca0ca7c4731e52a5.

audio:

Screenshot 2023-05-28 at 15 50 56

video:
Screenshot 2023-05-28 at 15 51 07

image:
Screenshot 2023-05-28 at 15 51 46

pdf corner:
Screenshot 2023-05-28 at 15 57 48

**Review by wxiaoguang — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Comment by silverwind** [comment]
This PR reduces gitea binary size by around 4MB.

Note that PDF.js would still be ideal for mobile support, but it's low-level API is too hard to work with for me at least. If there is a high-level wrapper that works in vanilla JS, we could still use it in the future to get back mobile support.

**Comment by silverwind** [comment]
PDFObject has a option `PDFJS_URL` which we could expose to users so that it can fall back to it:

https://pdfobject.com/examples/pdfjs.html

**Pull request merged**